src/ChannelServer/World/Inventory.cs

```diff
@@ -217,12 +279,30 @@ public int AddSilent(Item item)
 		/// <return>Index of the item.</return>
 		public int Add(Item item, InventoryAddType addType)
 		{
-			var index = this.AddSilent(item);
+			List<object> indexes = this.AddSilent(item);
+
+			object last_item = indexes.Last();
+			int last_index = (int)last_item.GetType().GetProperty("index").GetValue(last_item);
```

Comment:
Using reflection to access values is very inefficient. This will solve itself when you don't use anonymous objects anymore and can directly access the properties.